cmake/Modules/Findgflags.cmake

```diff
@@ -20,6 +20,11 @@ if (NOT gflags_FOUND)
   externalproject_add(gflags_gflags
       GIT_REPOSITORY ${URL}
       GIT_TAG        ${VERSION}
+      CMAKE_ARGS
+        -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
```

Comment:
Maybe it is also possible to define a variable like this and reuse it in different modules
```
set(DEPS_CMAKE_ARGS -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
        -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
        -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
        -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS})
```